samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -354,7 +355,19 @@ object SamzaContainer extends Logging {
 
     val chooserFactory = Util.getObj[MessageChooserFactory](chooserFactoryClassName)
 
-    val chooser = DefaultChooser(inputStreamMetadata, chooserFactory, config, samzaContainerMetrics.registry)
+    /**
+     * Constructs a map from SystemStreamPartition to number of tasks consuming from it.
+     * To mark a stream as bootstrapped, each SSP in the stream must be completely consumed
+     * by all the tasks that are assigned to it. This will help in enforcing the condition.
+     */
+    val sspToTaskCount: mutable.Map[SystemStreamPartition, Int] = mutable.Map[SystemStreamPartition, Int]()
+    for(taskModel <- containerModel.getTasks.values())  {
+      for (systemStreamPartition <- taskModel.getSystemStreamPartitions) {
+        sspToTaskCount.put(systemStreamPartition, sspToTaskCount.getOrElse(systemStreamPartition, 0) + 1)
+      }
+    }
+
+    val chooser = DefaultChooser(inputStreamMetadata, chooserFactory, config, samzaContainerMetrics.registry, sspToTaskCount.toMap)
```

Comment:
Something feels wrong here. Why should a chooser be concerned with which or how many tasks consume it?

I'll need to dig into the code to understand more, but I suspect that all the code for broadcast streams is on the task side, not the chooser side. If so, then we shouldn't be updating the chooser with broadcast logic.